docs/designs/ruby_test_plan_002.md

```diff
@@ -0,0 +1,150 @@
+# Test plan for RubyGems/Bundler package manager
+
+This document is intended to describe RubyGems/Bundler package manager
+usage patterns to be tested. The document follows package manager design
+proposed in [PR-565](https://github.com/containerbuildsystem/cachi2/pull/565).
+The document captures behaviors under test in a DSL for test scenario
+description. The format is straightforward and self-documenting, for a formal
+specification and tooling please refer to [the specification](https://cucumber.io/docs/gherkin/reference/).
+
+## General overview
+
+Cachi2 is intended to prefetch project's dependencies and to prepare a local
+environment to make it possible to build projects without network connectivity.
+Cachi2 can handle a number of package formats, RubyGems/Bundler (RG/B) is among
+the supported ones. Cachi2 does not support all features of RG/B, opting to
+support the most widespread and safe ones instead of supporting everything.
+
+The tests are to check that conforming packages could be preprocessed by Cachi2
+and that non-conforming packages are rejected by Cachi2.
+
+The checks are expected to be run against actual packages published either on
+rubygems.org, on github.com or locally.
+
+Each example in every test scenario requires a separate repository/package
+demonstrating an aspect under test. A list of necessary repositories could be
+found [below](#necessary-repositories).
+
+## Test cases
+
+@e2e_test
+Feature: Projects prepared with Cachi2 could be hermetically built
+    Scenario: a user can successfully build a project prepared with Cachi2
+        Given a clean repository of a well-defined project
+         When a user request to fetch dependencies for a well-defined project
+          And a user generates environment for well-defined project
+          And a user triggers a hermetic build
+         Then the build system exits successfully
+          And a build artifact is produced.
+
+
+@integration_test
+Feature: Cachi2 can prefetch dependencies for Ruby projects
+    
+    Scenario outline: a user wants to fetch dependencies for a package containing well-defined Gemfile.lock
+       Given  clean repository of a "well-defined project"
+        When  a user request to fetch dependencies for a "well-defined project"
+        Then  direct and transitive dependencies are fetched
+         And  dependencies could be found within the "well-defined project"
+    Examples:
+        |    well-defined project                          |
+        | Depends on gems from rubygems.org only           |
+        | Has correct .gemspec                             |
+        | Has correct Git dependency                       |
+        | Has several correct Git dependencies             |
+        | Has path dependency                              |
+        | Has several path dependencies                    |
+        | Has a mixture of gems, Git and path dependencies |
```

Comment:
I think the "well-defined" scenarios can be simplified to:
- Has a mixture of gems, gems from alternate registries, Git and path dependencies. Has a gemspec file.
- Does not have a gemspec file

Reply:
Makes sense, will update the doc in the next revision.